### ISSUE TITLE
Provide AWS Direct Connect Connection ID from packetfabric_cs_aws_hosted_connection

### ISSUE THREAD

**Issue by bpedman** [PacketFabric/terraform-provider-packetfabric#483]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When using the `packetfabric_cs_aws_hosted_connection` resource, I would like to have the AWS Direct Connect connection ID and VLAN ID returned from the resource so I can easily confirm the connection and create a VIF in AWS.

Note that this information is available from a similar provider for equinix so the information should be available to provide.

### New or Affected Resource(s)

* packetfabric_cs_aws_hosted_connection

### Potential Terraform Configuration

Current process
```hcl
# Create the hosted connection
resource "packetfabric_cs_aws_hosted_connection" "this" {
   ....
}

# From the AWS side: Accept the connection
# Wait for the connection to show up in AWS
resource "time_sleep" "wait_aws_connection" {
  create_duration = "1m"
  depends_on = [
    packetfabric_cs_aws_hosted_connection.this
  ]
}

# Retrieve the Direct Connect connections in AWS
data "aws_dx_connection" "this" {
  provider   = aws
  name       = packetfabric_cs_aws_hosted_connection.this.description
  depends_on = [time_sleep.wait_aws_connection]
}


# Sometimes, it takes more than 10min for the connection to become available
# Vote for below issue to get a timeout attribute added to the aws_dx_connection_confirmation resource
# https://github.com/hashicorp/terraform-provider-aws/issues/26335
resource "aws_dx_connection_confirmation" "this" {
  provider      = aws
  connection_id = data.aws_dx_connection.this.id

  lifecycle {
    ignore_changes = [
      connection_id
    ]
  }
}

# Create the vif...I can't remember but we may have had to do this in 2 runs to create the new connection to get the vlan ID after the fact, it would be great to have the vlan ID provided
resource "aws_dx_transit_virtual_interface" "this" {
  .....
  vlan             = var.aws_dx_vif_vlan_id # Due to a bug with aws_dx_connection where vlan_id is not properly returned we cannot use this value instead: data.aws_dx_connection.this.vlan_id
  .....
}
```

Proposed solution:

```hcl
# Create the hosted connection
resource "packetfabric_cs_aws_hosted_connection" "this" {
  ...
}

resource "aws_dx_connection_confirmation" "this" {
  connection_id = packetfabric_cs_aws_hosted_connection.this.aws_connection_id
}

resource "aws_dx_transit_virtual_interface" "this" {
  vlan             = packetfabric_cs_aws_hosted_connection.this.aws_vlan_id
}
```

This would be a lot easier to manage! :)

**Comment by rjouhann** [comment]
Hello @bpedman,

Thanks for your feedback. I have modified both `packetfabric_cs_aws_hosted_connection` and `packetfabric_cloud_router_connection_aws` resources and added 2 computed attributes `aws_connection_id` and `vlan_id_pf`.

This should be available in `v1.5.0` coming up next Wednesday. Do not hesitate to open a new issue if you have any other requests/feedback! Have a great weekend. 

Example output with a local build after modifying a bit the internal logic:
```
packetfabric_cs_aws_hosted_connection = {
  "account_uuid" = "789123"
  "aws_account_id" = "123456"
  "aws_connection_id" = "dxlag-fgl9ffux"
  "cloud_settings" = tolist([])
  "description" = "demo-simple-hare"
  "id" = "PF-CC-GOG-LAB-3440921-PF"
  "labels" = tolist([
    "terraform",
    "dev",
  ])
  "po_number" = "0123456"
  "pop" = "LAB1"
  "port" = "PF-AP-LAB4-3440735"
  "speed" = "50Mbps"
  "src_svlan" = tonumber(null)
  "timeouts" = null /* object */
  "vlan" = 107
  "vlan_id_pf" = 3002
  "zone" = "A"
}

packetfabric_cloud_router_connection_aws_crc_1 = {
  "account_uuid" = "789123"
  "aws_account_id" = "123456"
  "aws_connection_id" = "dxlag-fgl9ffux"
  "bgp_settings_uuid" = tostring(null)
  "circuit_id" = "PF-L3-CUST-3440734"
  "cloud_settings" = tolist([])
  "description" = "demo-simple-hare-LAB1"
  "id" = "PF-L3-CON-3440736"
  "is_public" = false
  "labels" = tolist([
    "terraform",
    "dev",
  ])
  "maybe_dnat" = false
  "maybe_nat" = false
  "po_number" = "0123456"
  "pop" = "LAB1"
  "published_quote_line_uuid" = tostring(null)
  "speed" = "50Mbps"
  "timeouts" = null /* object */
  "vlan_id_pf" = 3892
  "zone" = "A"
}
```

I will also be updating the use-case examples [here](https://github.com/PacketFabric/terraform-provider-packetfabric/tree/main/examples/use-cases) (look for *No Cloud Side Provisioning*)

**Comment by rjouhann** [comment]
Hello @bpedman, @nberkley-gp,

This should be part of the `v1.5.0` coming out tomorrow. We have added `cloud_provider_connection_id` and `vlan_id_pf` as compute attributes in `packetfabric_cs_aws_hosted_connection` and `packetfabric_cs_aws_hosted_connection` so they can be used to conform the connection and create the VIF without having to wait/use data-sources.

Example for AWS hosed Cloud resource:
```
packetfabric_cs_aws_hosted_connection = {
 "account_uuid" = "4b6382eb-e3d8-45a4-832f-fb559e17f3dd"
 "aws_account_id" = "687236962849"
 "cloud_provider_connection_id" = "dxcon-fh53d2yy"
 "cloud_settings" = tolist([])
 "description" = "demo-rich-narwhal"
 "id" = "PF-CC-GOG-LAB-3476477-PF"
 "labels" = tolist([
  "terraform",
  "dev",
 ])
 "po_number" = "0123456"
 "pop" = "LAB1"
 "port" = "PF-AP-LAB4-3476433"
 "speed" = "50Mbps"
 "src_svlan" = tonumber(null)
 "timeouts" = null /* object */
 "vlan" = 107
 "vlan_id_pf" = 3340
 "zone" = "A"
}
```
